src/main/webapp/scripts/species.js

```diff
@@ -0,0 +1,212 @@
+    // target = desired endpoint for pixelation
+    var target = pixel_factor.value;
+    // dx = speed of pixelation (# pixels/tick)
+    var dx = 10;
+    animated = true;
+    var underTarget = false;
+    doAnimation();
+
+    function doAnimation() {
+
+        if (PIXEL_FACTOR_CURR < target) {
+            PIXEL_FACTOR_CURR += dx;
+            // "Binary Search" appraoch to hone in on exact value since dx > 1
+            if (!underTarget) {
+                dx -= 1;
+            }
+            underTarget = true;
+        } else if (PIXEL_FACTOR_CURR > target) {
+            PIXEL_FACTOR_CURR -= dx;
+            // "Binary Search" appraoch to hone in on exact value since dx > 1
+            if (underTarget) {
+                dx -= 1;
+            }
+            underTarget = false;
+        } else {
+            animated = false;
+            return;
+        }
+
+        pixelate(PIXEL_FACTOR_CURR);
+
+        if (animated) {
+            requestAnimationFrame(doAnimation);
+        }
+    }
+}
+
+/** When mouse enters canvas, animate transition to original, full-resolution image.
+ *
+ * Zoom out to original, full-resolution image.
+ */
+ function animateFullResolution() {
+    // TODO: Change speed parameters to be a function of the original resolution.
+    // TODO: Ease functions?
+
+    // "speed" parameter. Must be different than parameter for other animation fn
+    // var dx = (originalNumPixels - PIXEL_FACTOR_CURR) / 100;
+    var dx = 2000;
+    animated = true;
+    doAnimation();
+
+    function doAnimation() {
+        PIXEL_FACTOR_CURR += dx;
+
+        if (PIXEL_FACTOR_CURR >= originalNumPixels) {
+            animated = false;
+            return;
+        }
+
+        pixelate(PIXEL_FACTOR_CURR);
+        
+        if (animated) {
+            requestAnimationFrame(doAnimation);
+        }
+    }
+}
+
+/** After mouse leaves canvas, animate transition to previous pixelation amount
+ *
+ * Zoom back in to old number of pixels.
+ */
+function animateOldResolution() {
+    // "speed" parameter. Must be different than parameter for other animation fn
+    // var dx = (PIXEL_FACTOR_CURR - PIXEL_FACTOR_OLD) / 50; 
+    var dx = 4000;
+
+    animated = true;
+    doAnimation();
+
+    function doAnimation() {
+        PIXEL_FACTOR_CURR -= dx;
+
+        if (PIXEL_FACTOR_CURR <= PIXEL_FACTOR_OLD) {
+            animated = false;
+            return;
+        }
+
+        pixelate(PIXEL_FACTOR_CURR);
+
+        if (animated) {
+            requestAnimationFrame(doAnimation);
+        }
+    }
+}
+
+// Stand-in code for older browsers that don't support the animation
+window.requestAnimationFrame = (function () {
+    return window.requestAnimationFrame || window.webkitRequestAnimationFrame || window.mozRequestAnimationFrame || function (callback) {
+        window.setTimeout(callback, 1000 / 60);
+    };
+})();
+
+// Test function that fetches sample JSON and modifies page to display information for one species
```

Comment:
I initially kept animation and data fetching together because they were both used only for `species.html` (hence `species.js`), but we are expanding data fetching functionality to other pages e.g. `gallery.html`, so I think it would be a good idea to separate them moving forward.